ARCHITECTURE.md

```diff
@@ -0,0 +1,131 @@
+# OpenSearch Security Plugin Architecture
+
+OpenSearch’s core systems do not include security features, these features are added by installing the Security Plugin.  The Security Plugin accesses OpenSearch’s internals to build up Authentication, Authorization, End to End Encryption, Audit Logging, and management interfaces.
+
+## Components
+
+The Security Plugin is packaged into a standard plugin zip file used by OpenSearch which can be installed by using the plugin tool.  The security configuration is accessible on disk for modification before the node has been turned on, afterward the admin tools or api endpoints need to be used.
+
+```mermaid
+graph TD
+    subgraph OpenSearch Node
+        subgraph File System
+            cfg[Security Configuration files]
+            adm[Admin Tools]
+        end
+        subgraph Indices
+            idx(Index 1..n)
+            secIdx[Security Index]
+        end
+        subgraph Plugins
+           pgns(Plugins 1..n)
+           sec[Security Plugin]
+        end
+
+        sec -- bootstrap security config --> cfg
+        sec -- refresh security config from cluster --> secIdx
+        adm -- backup/restore security config --> sec
+    end
+```
+
+### Security Plugin
+
+The runtime of the security plugin uses extension points to insert itself into the path actions.  Several security managemenet actions are registered in OpenSearch so they can be changed through REST API actions.
+
+### Security Configuration
+
+The security configuration is stored in an system index that is replicated to all nodes.  When a change has been made the Security plugin is reloaded to cleanly initialize its components with the new configuration.

```

Comment:
Is it replicated to all nodes? The cache is on all nodes, but I believe the security index itself may not have a replica on all of the nodes of a cluster.

Reply:
I pieced this explanation from how AbstractApiAction.saveAndUpdateConfigs / TransportConfigUpdateAction work.  Do you have changes you'd like to make, happy to look at a PR.